### ISSUE TITLE
live iso boot fail: ignition calling setfiles fails

### ISSUE THREAD

**Issue by dustymabe** [coreos/fedora-coreos-tracker#339]
Booting the [31.20191217.2.0](https://builds.coreos.fedoraproject.org/prod/streams/testing/builds/31.20191217.2.0/x86_64/fedora-coreos-31.20191217.2.0-live.x86_64.iso) ISO gives me an obscure failure. 

```
panic: interface conversion: interface {} is syscall.WaitStatus, not unix.WaitStatus
```

The obscure failure is [worked around](https://github.com/coreos/ignition/commit/1cfb161d9da4c334c89fd077ee5811e5c28f0878) in the newer version of ignition (not yet in an FCOS release). In order to get to the real failure I did a development build with the new ignition rpm (`ignition-2.1.1-1.git40c0b57.fc31.x86_64.rpm`) in it and re-ran the test. Now I get the real failure:

```
setfiles: Could not set context for /sysroot/home:  Read-only file system
```

<details>

```
Dec 19 20:15:07 localhost ignition[517]: Ignition 2.1.1
Dec 19 20:15:07 localhost ignition[517]: Stage: fetch
Dec 19 20:15:07 localhost ignition[517]: reading system config file "/usr/lib/ignition/base.ign"
Dec 19 20:15:07 localhost ignition[517]: parsing config with SHA512: ff6a5153be363997e4d5d3ea8cc4048373a457c48c4a5b134a08a30aacd167c1e0f099f0bdf1e24c99ad180628cd02b767b863b5fe3a8fce3fe1886847eb8e2e
Dec 19 20:15:07 localhost ignition[517]: parsed url from cmdline: ""
Dec 19 20:15:07 localhost ignition[517]: no config URL provided
Dec 19 20:15:07 localhost ignition[517]: reading system config file "/usr/lib/ignition/user.ign"
Dec 19 20:15:07 localhost ignition[517]: no config at "/usr/lib/ignition/user.ign"
Dec 19 20:15:07 localhost ignition[517]: noop provider fetching empty config
Dec 19 20:15:07 localhost ignition[517]: failed to fetch config: not a config (empty)
Dec 19 20:15:07 localhost ignition[517]: not a config (empty): ignoring user-provided config
Dec 19 20:15:07 localhost ignition[517]: fetch: fetch complete
Dec 19 20:15:07 localhost ignition[517]: fetch: fetch passed
Dec 19 20:15:07 localhost ignition[517]: Ignition finished successfully
Dec 19 20:15:07 localhost ignition[534]: Ignition 2.1.1
Dec 19 20:15:07 localhost ignition[534]: Stage: disks
Dec 19 20:15:07 localhost ignition[534]: reading system config file "/usr/lib/ignition/base.ign"
Dec 19 20:15:07 localhost ignition[534]: parsing config with SHA512: ff6a5153be363997e4d5d3ea8cc4048373a457c48c4a5b134a08a30aacd167c1e0f099f0bdf1e24c99ad180628cd02b767b863b5fe3a8fce3fe1886847eb8e2e
Dec 19 20:15:07 localhost ignition[534]: parsed url from cmdline: ""
Dec 19 20:15:07 localhost ignition[534]: no config URL provided
Dec 19 20:15:07 localhost ignition[534]: reading system config file "/usr/lib/ignition/user.ign"
Dec 19 20:15:07 localhost ignition[534]: no config at "/usr/lib/ignition/user.ign"
Dec 19 20:15:07 localhost ignition[534]: noop provider fetching empty config
Dec 19 20:15:07 localhost ignition[534]: failed to fetch config: not a config (empty)
Dec 19 20:15:07 localhost ignition[534]: not a config (empty): ignoring user-provided config
Dec 19 20:15:07 localhost ignition[534]: disks: disks passed
Dec 19 20:15:07 localhost ignition[534]: Ignition finished successfully
Dec 19 20:15:16 localhost ignition[772]: INFO     : Ignition 2.1.1
Dec 19 20:15:16 localhost ignition[772]: INFO     : Stage: mount
Dec 19 20:15:16 localhost ignition[772]: INFO     : reading system config file "/usr/lib/ignition/base.ign"
Dec 19 20:15:16 localhost ignition[772]: DEBUG    : parsing config with SHA512: ff6a5153be363997e4d5d3ea8cc4048373a457c48c4a5b134a08a30aacd167c1e0f099f0bdf1e24c99ad180628cd02b767b863b5fe3a8fce3fe1886847eb8e2e
Dec 19 20:15:16 localhost ignition[772]: DEBUG    : parsed url from cmdline: ""
Dec 19 20:15:16 localhost ignition[772]: INFO     : no config URL provided
Dec 19 20:15:16 localhost ignition[772]: INFO     : reading system config file "/usr/lib/ignition/user.ign"
Dec 19 20:15:16 localhost ignition[772]: INFO     : no config at "/usr/lib/ignition/user.ign"
Dec 19 20:15:16 localhost ignition[772]: DEBUG    : noop provider fetching empty config
Dec 19 20:15:16 localhost ignition[772]: WARNING  : failed to fetch config: not a config (empty)
Dec 19 20:15:16 localhost ignition[772]: INFO     : not a config (empty): ignoring user-provided config
Dec 19 20:15:16 localhost ignition[772]: INFO     : mount: mount passed
Dec 19 20:15:16 localhost ignition[772]: INFO     : Ignition finished successfully
Dec 19 20:15:17 localhost ignition[818]: INFO     : Ignition 2.1.1
Dec 19 20:15:17 localhost ignition[818]: INFO     : Stage: files
Dec 19 20:15:17 localhost ignition[818]: INFO     : reading system config file "/usr/lib/ignition/base.ign"
Dec 19 20:15:17 localhost ignition[818]: DEBUG    : parsing config with SHA512: ff6a5153be363997e4d5d3ea8cc4048373a457c48c4a5b134a08a30aacd167c1e0f099f0bdf1e24c99ad180628cd02b767b863b5fe3a8fce3fe1886847eb8e2e
Dec 19 20:15:17 localhost ignition[818]: DEBUG    : parsed url from cmdline: ""
Dec 19 20:15:17 localhost ignition[818]: INFO     : no config URL provided
Dec 19 20:15:17 localhost ignition[818]: INFO     : reading system config file "/usr/lib/ignition/user.ign"
Dec 19 20:15:17 localhost ignition[818]: INFO     : no config at "/usr/lib/ignition/user.ign"
Dec 19 20:15:17 localhost ignition[818]: DEBUG    : noop provider fetching empty config
Dec 19 20:15:17 localhost ignition[818]: WARNING  : failed to fetch config: not a config (empty)
Dec 19 20:15:17 localhost ignition[818]: INFO     : not a config (empty): ignoring user-provided config
Dec 19 20:15:17 localhost ignition[818]: INFO     : files: createUsers: op(1): [started]  creating or modifying user "core"
Dec 19 20:15:17 localhost ignition[818]: DEBUG    : files: createUsers: op(1): executing: "useradd" "--root" "/sysroot" "--create-home" "--password" "*" "--comment" "CoreOS Admin" "--groups" "adm,sudo,systemd-journal,wheel" "core"
Dec 19 20:15:17 localhost ignition[818]: INFO     : files: createUsers: op(1): [finished] creating or modifying user "core"
Dec 19 20:15:17 localhost ignition[818]: INFO     : files: op(2): [started]  relabeling 17 patterns
Dec 19 20:15:17 localhost ignition[818]: DEBUG    : files: op(2): executing: "setfiles" "-vFi0" "-r" "/sysroot" "/sysroot/etc/selinux/targeted/contexts/files/file_contexts" "-f" "-"
Dec 19 20:15:17 localhost ignition[818]: CRITICAL : files: op(2): [failed]   relabeling 17 patterns: exit status 255: Cmd: "setfiles" "-vFi0" "-r" "/sysroot" "/sysroot/etc/selinux/targeted/contexts/files/file_contexts" "-f" "-" Stdout: "Relabeled /sysroot/etc/passwd from unlabeled to system_u:object_r:passwd_file_t:s0\nRelabeled /sysroot/etc/passwd- from unlabeled to system_u:object_r:passwd_file_t:s0\nRelabeled /sysroot/etc/group from unlabeled to system_u:object_r:pas
swd_file_t:s0\nRelabeled /sysroot/etc/group- from unlabeled to system_u:object_r:passwd_file_t:s0\nRelabeled /sysroot/etc/shadow from unlabeled to system_u:object_r:shadow_t:s0\nRelabeled /sysroot/etc/shadow- from unlabeled to system_u:object_r:shadow_t:s0\nRelabeled /sysroot/etc/gshadow from unlabeled to system_u:object_r:shadow_t:s0\nRelabeled /sysroot/etc/gshadow- from unlabeled to system_u:object_r:shadow_t:s0\nRelabeled /sysroot/etc/subuid from unlabeled to system_
u:object_r:etc_t:s0\nRelabeled /sysroot/etc/subuid- from unlabeled to system_u:object_r:etc_t:s0\nRelabeled /sysroot/etc/subgid from unlabeled to system_u:object_r:etc_t:s0\nRelabeled /sysroot/etc/subgid- from unlabeled to system_u:object_r:etc_t:s0\nRelabeled /sysroot/etc/.pwd.lock from unlabeled to system_u:object_r:passwd_file_t:s0\nRelabeled /sysroot/var/home from unlabeled to system_u:object_r:home_root_t:s0\nRelabeled /sysroot/var/home/core from unlabeled to uncon
fined_u:object_r:user_home_dir_t:s0\nRelabeled /sysroot/var/home/core/.bashrc from unlabeled to unconfined_u:object_r:user_home_t:s0\nRelabeled /sysroot/var/home/core/.bash_profile from unlabeled to unconfined_u:object_r:user_home_t:s0\nRelabeled /sysroot/var/home/core/.bash_logout from unlabeled to unconfined_u:object_r:user_home_t:s0\nRelabeled /sysroot/var/roothome from unlabeled to system_u:object_r:admin_home_t:s0\nRelabeled /sysroot/var/roothome/.bashrc from unlab
eled to system_u:object_r:admin_home_t:s0\nRelabeled /sysroot/var/roothome/.bash_profile from unlabeled to system_u:object_r:admin_home_t:s0\nRelabeled /sysroot/var/roothome/.bash_logout from unlabeled to system_u:object_r:admin_home_t:s0\n" Stderr: "setfiles: Could not set context for /sysroot/home:  Read-only file system\nsetfiles: Could not set context for /sysroot/root:  Read-only file system\n"
Dec 19 20:15:17 localhost ignition[818]: files failedFull config:
Dec 19 20:15:17 localhost ignition[818]: {
Dec 19 20:15:17 localhost ignition[818]:   "ignition": {
Dec 19 20:15:17 localhost ignition[818]:     "config": {
Dec 19 20:15:17 localhost ignition[818]:       "replace": {
Dec 19 20:15:17 localhost ignition[818]:         "source": null,
Dec 19 20:15:17 localhost ignition[818]:         "verification": {}
Dec 19 20:15:17 localhost ignition[818]:       }
Dec 19 20:15:17 localhost ignition[818]:     },
Dec 19 20:15:17 localhost ignition[818]:     "proxy": {},
Dec 19 20:15:17 localhost ignition[818]:     "security": {
Dec 19 20:15:17 localhost ignition[818]:       "tls": {}
Dec 19 20:15:17 localhost ignition[818]:     },
Dec 19 20:15:17 localhost ignition[818]:     "timeouts": {}
Dec 19 20:15:17 localhost ignition[818]:   },
Dec 19 20:15:17 localhost ignition[818]:   "passwd": {
Dec 19 20:15:17 localhost ignition[818]:     "users": [
Dec 19 20:15:17 localhost ignition[818]:       {
Dec 19 20:15:17 localhost ignition[818]:         "gecos": "CoreOS Admin",
Dec 19 20:15:17 localhost ignition[818]:         "groups": [
Dec 19 20:15:17 localhost ignition[818]:           "adm",
Dec 19 20:15:17 localhost ignition[818]:           "sudo",
Dec 19 20:15:17 localhost ignition[818]:           "systemd-journal",
Dec 19 20:15:17 localhost ignition[818]:           "wheel"
Dec 19 20:15:17 localhost ignition[818]:         ],
Dec 19 20:15:17 localhost ignition[818]:         "name": "core"
Dec 19 20:15:17 localhost ignition[818]:       }
Dec 19 20:15:17 localhost ignition[818]:     ]
Dec 19 20:15:17 localhost ignition[818]:   },
Dec 19 20:15:17 localhost ignition[818]:   "storage": {},
Dec 19 20:15:17 localhost ignition[818]:   "systemd": {}
Dec 19 20:15:17 localhost ignition[818]: }CRITICAL : Ignition failed: failed to handle relabeling: exit status 255: Cmd: "setfiles" "-vFi0" "-r" "/sysroot" "/sysroot/etc/selinux/targeted/contexts/files/file_contexts" "-f" "-" Stdout: "Relabeled /sysroot/etc/passwd from unlabeled to system_u:object_r:passwd_file_t:s0\nRelabeled /sysroot/etc/passwd- from unlabeled to system_u:object_r:passwd_file_t:s0\nRelabeled /sysroot/etc/group from unlabeled to system_u:object_r:passw
d_file_t:s0\nRelabeled /sysroot/etc/group- from unlabeled to system_u:object_r:passwd_file_t:s0\nRelabeled /sysroot/etc/shadow from unlabeled to system_u:object_r:shadow_t:s0\nRelabeled /sysroot/etc/shadow- from unlabeled to system_u:object_r:shadow_t:s0\nRelabeled /sysroot/etc/gshadow from unlabeled to system_u:object_r:shadow_t:s0\nRelabeled /sysroot/etc/gshadow- from unlabeled to system_u:object_r:shadow_t:s0\nRelabeled /sysroot/etc/subuid from unlabeled to system_u:
object_r:etc_t:s0\nRelabeled /sysroot/etc/subuid- from unlabeled to system_u:object_r:etc_t:s0\nRelabeled /sysroot/etc/subgid from unlabeled to system_u:object_r:etc_t:s0\nRelabeled /sysroot/etc/subgid- from unlabeled to system_u:object_r:etc_t:s0\nRelabeled /sysroot/etc/.pwd.lock from unlabeled to system_u:object_r:passwd_file_t:s0\nRelabeled /sysroot/var/home from unlabeled to system_u:object_r:home_root_t:s0\nRelabeled /sysroot/var/home/core from unlabeled to unconfi
ned_u:object_r:user_home_dir_t:s0\nRelabeled /sysroot/var/home/core/.bashrc from unlabeled to unconfined_u:object_r:user_home_t:s0\nRelabeled /sysroot/var/home/core/.bash_profile from unlabeled to unconfined_u:object_r:user_home_t:s0\nRelabeled /sysroot/var/home/core/.bash_logout from unlabeled to unconfined_u:object_r:user_home_t:s0\nRelabeled /sysroot/var/roothome from unlabeled to system_u:object_r:admin_home_t:s0\nRelabeled /sysroot/var/roothome/.bashrc from unlabel
ed to system_u:object_r:admin_home_t:s0\nRelabeled /sysroot/var/roothome/.bash_profile from unlabeled to system_u:object_r:admin_home_t:s0\nRelabeled /sysroot/var/roothome/.bash_logout from unlabeled to system_u:object_r:admin_home_t:s0\n" Stderr: "setfiles: Could not set context for /sysroot/home:  Read-only file system\nsetfiles: Could not set context for /sysroot/root:  Read-only file system\n"
Dec 19 20:15:17 localhost ignition[833]: INFO     : Ignition 2.1.1
Dec 19 20:15:17 localhost ignition[833]: INFO     : Stage: umount
Dec 19 20:15:17 localhost ignition[833]: INFO     : reading system config file "/usr/lib/ignition/base.ign"
Dec 19 20:15:17 localhost ignition[833]: DEBUG    : parsing config with SHA512: ff6a5153be363997e4d5d3ea8cc4048373a457c48c4a5b134a08a30aacd167c1e0f099f0bdf1e24c99ad180628cd02b767b863b5fe3a8fce3fe1886847eb8e2e
Dec 19 20:15:17 localhost ignition[833]: DEBUG    : parsed url from cmdline: ""
Dec 19 20:15:17 localhost ignition[833]: INFO     : no config URL provided
Dec 19 20:15:17 localhost ignition[833]: INFO     : reading system config file "/usr/lib/ignition/user.ign"
Dec 19 20:15:17 localhost ignition[833]: INFO     : no config at "/usr/lib/ignition/user.ign"
Dec 19 20:15:17 localhost ignition[833]: DEBUG    : noop provider fetching empty config
Dec 19 20:15:17 localhost ignition[833]: WARNING  : failed to fetch config: not a config (empty)
Dec 19 20:15:17 localhost ignition[833]: INFO     : not a config (empty): ignoring user-provided config
Dec 19 20:15:17 localhost ignition[833]: INFO     : umount: umount passed
Dec 19 20:15:17 localhost ignition[833]: INFO     : Ignition finished successfully
```

</details>

**Comment by jlebon** [comment]
Hmm, right. `/home` and `/root` are symlinks on the ro sysroot so it can't relabel those directly. But actually, they should already be correctly labeled from disk creation time. So we should be able to make Ignition skip over those.

**Comment by dustymabe** [comment]
I believe this can be closed now since `ignition-2.1.1-3.git40c0b57.fc31.x86_64` is in the latest released content.
